Segmentation fault when deploying a malformed `docker-compose.yml`

**Describe the bug**
A panic occurs when running `okteto deploy -f ./docker-compose.yml` on a malformed `docker-compose.yml`, example:

```yml
version: "3"

services:
  web:
    build: .
    ports:
      - 8080:8080
  another-service:

  mongodb:
    image: bitnami/mongodb:latest
    ports:
      - 27017
    volumes:
      - mongodb-data:/bitnami/mongodb

volumes:
  mongodb-data:
```

Notice how `another-service:` is empty

**To Reproduce**
Steps to reproduce the behavior:

1. Clone https://github.com/okteto/go-api-docker-compose
1. Replace the `docker-compose.yml` with the one provided above
1. Run `okteto deploy -f ./docker-compose.yml`
1. Notice the panic error

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x478 pc=0x1011c3e18]

goroutine 30 [running]:
gopkg.in/yaml%2ev2.handleErr(0x14001887208)
	gopkg.in/yaml.v2@v2.4.0/yaml.go:249 +0x80
panic({0x102c8f3a0, 0x1047af330})
	runtime/panic.go:884 +0x204
github.com/okteto/okteto/pkg/model.validateExtensions({{0x104716cb8, 0x1}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, 0x140018758f0, 0x0, ...})
	github.com/okteto/okteto/pkg/model/stack_serializer.go:1627 +0x1c8
github.com/okteto/okteto/pkg/model.(*Stack).UnmarshalYAML(0x14000a5e600, 0x14001904160)
	github.com/okteto/okteto/pkg/model/stack_serializer.go:293 +0x7c
gopkg.in/yaml%2ev2.(*decoder).callUnmarshaler(0x140002e1920, 0x140018f8f50, {0x12cea2640, 0x14000a5e600})
	gopkg.in/yaml.v2@v2.4.0/decode.go:270 +0xa0
gopkg.in/yaml%2ev2.(*decoder).prepare(0x8?, 0x140018f8f50, {0x102f75b80?, 0x14000a5e600?, 0x1?})
	gopkg.in/yaml.v2@v2.4.0/decode.go:313 +0x260
gopkg.in/yaml%2ev2.(*decoder).unmarshal(0x140002e1920, 0x140018f8f50, {0x102f75b80?, 0x14000a5e600?, 0x14000100000?})
	gopkg.in/yaml.v2@v2.4.0/decode.go:364 +0x108
gopkg.in/yaml%2ev2.(*decoder).document(0x140018f8ee0?, 0x140018f8f50?, {0x102f75b80?, 0x14000a5e600?, 0x14001902600?})
	gopkg.in/yaml.v2@v2.4.0/decode.go:384 +0x7c
gopkg.in/yaml%2ev2.(*decoder).unmarshal(0x102de8d60?, 0x14000a5e600?, {0x102f75b80?, 0x14000a5e600?, 0x0?})
	gopkg.in/yaml.v2@v2.4.0/decode.go:360 +0x1ac
gopkg.in/yaml%2ev2.unmarshal({0x1400061b400, 0xed, 0x100}, {0x102de8d60?, 0x14000a5e600}, 0x1)
	gopkg.in/yaml.v2@v2.4.0/yaml.go:148 +0x3e0
gopkg.in/yaml%2ev2.UnmarshalStrict(...)
	gopkg.in/yaml.v2@v2.4.0/yaml.go:89
github.com/okteto/okteto/pkg/model.ReadStack({0x140015ffc00, 0xf0, 0x200}, 0x1)
	github.com/okteto/okteto/pkg/model/stack.go:312 +0xac
github.com/okteto/okteto/pkg/model.GetStackFromPath({0x0, 0x0}, {0x140015fd2c0, 0x41}, 0xe8?)
	github.com/okteto/okteto/pkg/model/stack.go:230 +0x1ac
github.com/okteto/okteto/pkg/model.getStack({0x0, 0x0}, {0x140015fd2c0, 0x41})
	github.com/okteto/okteto/pkg/model/stack.go:764 +0x12c
github.com/okteto/okteto/pkg/model.LoadStack({0x0, 0x0}, {0x1400187ff70?, 0x1, 0x140018878d8?}, 0x0)
	github.com/okteto/okteto/pkg/model/stack.go:727 +0x11c
github.com/okteto/okteto/pkg/model.getManifestFromFile({0x14000052044, 0x2e}, {0x140015fd2c0, 0x41})
	github.com/okteto/okteto/pkg/model/manifest.go:383 +0x390
github.com/okteto/okteto/pkg/model.getManifestFromDevFilePath({0x14000052044, 0x2e}, {0x1400044b200?, 0x0?})
	github.com/okteto/okteto/pkg/model/manifest.go:245 +0xe8
github.com/okteto/okteto/pkg/model.GetManifestV2({0x1400044b200, 0x12})
	github.com/okteto/okteto/pkg/model/manifest.go:288 +0x3c
github.com/okteto/okteto/cmd/deploy.(*DeployCommand).RunDeploy(0x1400078f800, {0x1030b3c08, 0x14000122000}, 0x1400159e5b0)
	github.com/okteto/okteto/cmd/deploy/deploy.go:302 +0x238
github.com/okteto/okteto/cmd/deploy.Deploy.func1.1()
	github.com/okteto/okteto/cmd/deploy/deploy.go:206 +0x64
created by github.com/okteto/okteto/cmd/deploy.Deploy.func1
	github.com/okteto/okteto/cmd/deploy/deploy.go:205 +0xa20
```


**Expected behavior**
The CLI should fail gracefully similar to:

```
 x  Invalid manifest:
Unmarshal error: 'yaml: unmarshal errors:
  line 1: cannot unmarshal !!str `XXX` into model.devType'
```

--- a/pkg/errors/errors.go
+++ b/pkg/errors/errors.go
@@ -191,7 +191,7 @@ var (
 	ErrInvalidManifest = errors.New("invalid manifest")
 
 	// ErrServiceEmpty is raised whenever service is empty in docker-compose
-	ErrServiceEmpty = errors.New("Service cannot be empty")
+	ErrServiceEmpty = errors.New("service cannot be empty")
 
 	// ErrEmptyManifest is raised when cannot detected content to read in manifest
 	ErrEmptyManifest = errors.New("no content detected for okteto.yml file")

--- a/pkg/errors/errors.go
+++ b/pkg/errors/errors.go
@@ -190,7 +190,7 @@ var (
 	// ErrInvalidManifest is raised when cannot unmarshal manifest properly
 	ErrInvalidManifest = errors.New("invalid manifest")
 
-	//  ErrServiceEmpty is raised whenever service is empty in docker-compose
+	// ErrServiceEmpty is raised whenever service is empty in docker-compose
 	ErrServiceEmpty = errors.New("Service cannot be empty")
 
 	// ErrEmptyManifest is raised when cannot detected content to read in manifest

--- a/pkg/errors/errors.go
+++ b/pkg/errors/errors.go
@@ -190,6 +190,9 @@ var (
 	// ErrInvalidManifest is raised when cannot unmarshal manifest properly
 	ErrInvalidManifest = errors.New("invalid manifest")
 
+	//  ErrServiceEmpty is raised whenever service is empty in docker-compose
+	ErrServiceEmpty = errors.New("Service cannot be empty")
+
 	// ErrEmptyManifest is raised when cannot detected content to read in manifest
 	ErrEmptyManifest = errors.New("no content detected for okteto.yml file")
 

--- a/pkg/model/manifest.go
+++ b/pkg/model/manifest.go
@@ -417,8 +417,7 @@ func getManifestFromFile(cwd, manifestPath string) (*Manifest, error) {
 			if errors.Is(stackErr, errDependsOn) {
 				return nil, stackErr
 			}
-			// if not return original manifest err
-			return nil, err
+			return nil, stackErr
 		}
 		stackManifest.Deploy.ComposeSection.Stack = s
 		if stackManifest.Deploy.ComposeSection.Stack.Name != "" {

--- a/pkg/model/manifest.go
+++ b/pkg/model/manifest.go
@@ -417,7 +417,12 @@ func getManifestFromFile(cwd, manifestPath string) (*Manifest, error) {
 			if errors.Is(stackErr, errDependsOn) {
 				return nil, stackErr
 			}
-			return nil, stackErr
+
+			if strings.Contains(stackErr.Error(), oktetoErrors.ErrServiceEmpty.Error()) {
+				return nil, stackErr
+			}
+			// if not return original manifest err
+			return nil, err
 		}
 		stackManifest.Deploy.ComposeSection.Stack = s
 		if stackManifest.Deploy.ComposeSection.Stack.Name != "" {

--- a/pkg/model/manifest.go
+++ b/pkg/model/manifest.go
@@ -418,7 +418,7 @@ func getManifestFromFile(cwd, manifestPath string) (*Manifest, error) {
 				return nil, stackErr
 			}
 
-			if strings.Contains(stackErr.Error(), oktetoErrors.ErrServiceEmpty.Error()) {
+			if errors.Is(stackErr, oktetoErrors.ErrServiceEmpty) {
 				return nil, stackErr
 			}
 			// if not return original manifest err

--- a/pkg/model/manifest_test.go
+++ b/pkg/model/manifest_test.go
@@ -843,6 +843,14 @@ func Test_getManifestFromFile(t *testing.T) {
     image: test`),
 		},
 		{
+			name:          "manifestPath to a invalid compose file with empty service",
+			manifestBytes: nil,
+			composeBytes: []byte(`services:
+  test:
+          `),
+			expectedErr: oktetoErrors.ErrServiceEmpty,
+		},
+		{
 			name:          "manifestPath to empty okteto manifest, no compose file",
 			manifestBytes: []byte(``),
 			composeBytes:  nil,

--- a/pkg/model/stack.go
+++ b/pkg/model/stack.go
@@ -347,6 +347,9 @@ func ReadStack(bytes []byte, isCompose bool) (*Stack, error) {
 			_, _ = sb.WriteString("    See https://okteto.com/docs/reference/compose/ for details")
 			return nil, errors.New(sb.String())
 		}
+		if errors.Is(err, oktetoErrors.ErrServiceEmpty) {
+			return nil, err
+		}
 
 		msg := strings.Replace(err.Error(), "yaml: unmarshal errors:", "invalid compose manifest:", 1)
 		msg = strings.TrimSuffix(msg, "in type model.Stack")

--- a/pkg/model/stack_serializer.go
+++ b/pkg/model/stack_serializer.go
@@ -1629,7 +1629,7 @@ func validateExtensions(stack StackRaw) error {
 
 	for svcName, svc := range stack.Services {
 		if svc == nil {
-			return fmt.Errorf("%s: %w", oktetoErrors.ErrInvalidManifest, fmt.Errorf("Service %s is empty", svcName))
+			return fmt.Errorf("%s: %w", oktetoErrors.ErrInvalidManifest, oktetoErrors.ErrServiceEmpty)
 		}
 		for extension := range svc.Extensions {
 			nonValidFields = append(nonValidFields, fmt.Sprintf("services[%s].%s", svcName, extension))

--- a/pkg/model/stack_serializer.go
+++ b/pkg/model/stack_serializer.go
@@ -26,6 +26,7 @@ import (
 
 	"github.com/kballard/go-shellquote"
 	"github.com/okteto/okteto/pkg/cache"
+	oktetoErrors "github.com/okteto/okteto/pkg/errors"
 	"github.com/okteto/okteto/pkg/filesystem"
 	"github.com/okteto/okteto/pkg/model/forward"
 	apiv1 "k8s.io/api/core/v1"
@@ -1627,6 +1628,9 @@ func validateExtensions(stack StackRaw) error {
 	}
 
 	for svcName, svc := range stack.Services {
+		if svc == nil {
+			return fmt.Errorf("%s: %w", oktetoErrors.ErrInvalidManifest, fmt.Errorf("Service %s is empty", svcName))
+		}
 		for extension := range svc.Extensions {
 			nonValidFields = append(nonValidFields, fmt.Sprintf("services[%s].%s", svcName, extension))
 		}

--- a/pkg/model/stack_serializer_test.go
+++ b/pkg/model/stack_serializer_test.go
@@ -886,6 +886,12 @@ func Test_validateCommandArgs(t *testing.T) {
 			expectedErr: false,
 		},
 		{
+			name:        "COMPOSE-service-empty",
+			manifest:    []byte("services:\n  app:\n"),
+			isCompose:   true,
+			expectedErr: true,
+		},
+		{
 			name:        "STACK-only-entrypoint",
 			manifest:    []byte("services:\n  app:\n    entrypoint: [\"/usr/bin/rpk\", \"redpanda\"]\n    image: okteto/vote:1"),
 			isCompose:   false,

